### DISCUSSION TITLE
ESP32Cx versions do not compile at my desk

### DISCUSSION THREAD

**Discussion by Mape24** [arendst/Tasmota#22190] [Support request]
At Building with this setting **_env:tasmotaESP32c..._** I get this bug:

`riscv32-esp-elf-g++: fatal error: cannot execute 'c:/users/moppel/.platformio/packages/toolchain-riscv32-esp/bin/../lib/gcc/riscv32-esp-elf/12.2.0/../../../../riscv32-esp-elf/bin/as.exe': CreateProcess: No such file or directory`

- The mentioned folder does exist!
- error appears at first call of _riscv32-esp-elf\bin\as.exe_ 
- other tools like _bin\riscv32-esp-elf-c++.exe_ run successfully before this error
- no problem with _env:tasmotaESP32s2_  or with _env:tasmota_ target
- no problem with other (none Tasmota) PlatformIO example using ESP32C3

I have actually not found , where the tool folders are defined and configured. Can you help?

Regards Marten

**Comment by Jason2866** [comment]
Your Platformio setup seems broken. Close VSC. Delete hidden folder `.platformio` Start VSC. Wait, Platformio does rebuild. Do the restart when you asked for.

**Comment by Mape24** [comment]
Yes, I did that already. I have red this hint before. It took me 1 hour, but it did not help.
However, I have a fresh setup now.

My PlatformIO works well for other PlatformIO projects using ESP32C3.

**Comment by Jason2866** [comment]
Is your system Windows based? In this case you run probably in the Windows path length limit.
You may can workaround by uncomment the lines 11 - 16 in `platformio_override.ini`

**Comment by Mape24** [comment]
Thank you! Windows path length limit was the reason.
Your suggestion to modify `platformio_override.ini` puts the built folder into c:/  root folder in order to keep the path names short.
```
core_dir = .platformio
workspace_dir = c:\.pio
```
(new installation of pio takes 30 min)

Probably there is another solution.
Start windoes REGEDIT and select
`HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Control\FileSystem`
set `LongPathsEnabled. 1`

This I havent tried yet.
